Refreshed gem

bitaxis opened the pull request:
I refreshed the gem with updated dependency versions, added Rails 5.0 & 5.1 to the appraisal, and use Ruby 2.4.1 as well.

All tests and appraisals pass.

bitaxis committed:
Use Ruby 2.4.1

bitaxis committed:
Update dependencies.

bitaxis committed:
Version bump.

bitaxis committed:
Add missing require.

bitaxis committed:
1. Add appraisal for Rails 5.1
2. Lower Rails run-time dependency back down to 3.2

bitaxis committed:
Replace 5.0.0.rc1 with 5.0 proper.

bitaxis committed:
Merge branch 'release/1.1.0'

bitaxis committed:
Merge tag '1.1.0' into develop

Refresh gem with updated dependencies, Ruby 2.4.1, and Rails 5.1 compatibility.

bitaxis committed:
Raise error by default in empty migration.

bitaxis committed:
Add #execute_operation and #execute_suboperation.

bitaxis committed:
Make #execute_operation and #execute_suboperation public.

bitaxis committed:
1. Make execute helpers private.
2. Invoke it with either execute_[sub]operation(cql) or execute_[sub]operation(message, cql).

bitaxis committed:
Merge branch 'feature/add-execute-operation-helpers' into develop

bitaxis committed:
Version bump.

bitaxis committed:
1. Add #execute_file.
2. Fine-tune #execute_[sub]operation.

bitaxis committed:
1. Add #execute_text.
2. Rename #execute_operation to #execute_statement.
3. Remove #execute_suboperation.

bitaxis committed:
Fix typo.

bitaxis committed:
Merge branch 'release/1.2.0'

bitaxis committed:
Merge tag '1.2.0' into develop

1. Change template to raise error by default.
2. Add helper methods.

hsgubert commented:
Hi @bitaxis, interesting contributions. 
Is there anything you could contribute/update on the README file? Also, it seems to me the wiki page https://github.com/hsgubert/cassandra_migrations/wiki/Migrations-DSL could be updated with your new helpers.

Just for consistency with previous contributions, could you add your name to the Contributors section on the README file instead of altering the .gemspec?  In case you with to do bigger contributions and take the responsability to maintain the gem on the long term I would have no problem in adding you to the authors list in the future.

bitaxis commented:
Sure.  I'll take a look at the Wiki pages and add my name to the list of contributors, and make improvements, should my use of it continues.  However, I am not at a place where I would want to take over the gem's maintenance.

hsgubert commented:
no problem @bitaxis . I just mentioned about gem maintenance in case you wanted to be included in the authors list. Your contribution is very welcome in any case! (and in any size)

bitaxis committed:
1. Improve !using_keyspace by adding begin ... ensure ... end.
2. Add !use_keyspace so that the migration needn't change with environments.
3. Improve !use by converting keyspace to string in case it's a symbol.

bitaxis committed:
Expose use_keyspace to migrations.

bitaxis committed:
Merge branch 'feature/improve-other-keyspace-support' into develop

bitaxis committed:
Version bump.

bitaxis committed:
1. Add comment for specific test that requires real Cassandra in order to work, for now.
2. Add spec for #use_keyspace.

bitaxis committed:
1. Ignore .byebug_history.
2. Add additional keyspace environments for testing.
3. Add spec for .use_keyspace method.
4. Refine comments for the .use_keyspace method.

bitaxis committed:
Latest changes.

bitaxis committed:
Remove need to have real Cassandra instance when checking that using a non-existent keyspace would result in an exception being raised.

bitaxis committed:
Merge branch 'release/1.3.0'

bitaxis committed:
Forgot to update release date in gemspec.

bitaxis committed:
Merge branch 'hotfix/1.3.1'

bitaxis commented:
@hsgubert, I updated the Wiki on your repo to match version 1.2.0 of the gem on my fork.

However, I have since released version 1.3.1 of the gem on my fork, which contains functionality that does not exist in yours.  Hence, I also cloned your Wiki and made changes [there](https://github.com/bitaxis/cassandra_migrations/wiki).  Please take a look and satisfy this pull request as well as merging the Wiki changes I made back to yours.

hsgubert committed:
Update cassandra_migrations.gemspec

For now the best place to put your name and contact is on the README.md file as a contributor (also to be fair with other contributors). As I said earlier, if you wish to help maintain the gem on the long term I see no problem in putting you as an author too.

hsgubert committed:
Adding @bitaxis as contributor

hsgubert reviewed (changes requested):
You pull request is 90% ok. I just didn't find unit tests for the new methods in migration.rb.

Also, I know you improved the wiki pages, but please, do review the README.md file to see if all instructions there still work and the versions of dependencies are still correct. I noticed you had forgotten to put yourself as a contributor there, so I am not so confident you reviewed the README.md file.

Address those issues and we can merge your commits and publish a new version of the gem. 

Thank you for your help in this project.

bitaxis commented:
@hsgubert, thank you for acknowledging my contributions.

I took a look at the existing specs, and I don't see tests for previously existing methods such as `announce_migration`, `announce_operation`, and `announce_suboperation`.  If you can provde some guidance/insight on how I might test them, I'd be happy to follow along.

Also, having reviewed the `README.md` file, I don't think I've changed anything that would cause existing features to break.

hsgubert commented:
I see your point. I think the reason why we never were really strict about the **announce_** methods is because they just print statements. But it wouldn't hurt to have tests for them as well, just to ensure no crash happens when they're called.

About the new **execute_** methods you created, they have business logic in them and therefore should be tested. We just need 3 test cases:

1. For the `execute_text` method, just pass a few examples of CQL text (including trailing spaces or not, semicolon or not) and check that `execute` is called with the expected parameter (you can use rspec-mocks for that)

2. `execute_statement`: just test it delegates call to `execute`

3. `execute_file`: stub File.read and make it return a CQL text when a certain argument is received. Then just check `execute` is called with the correct parameter

Something like that is enough. If you want to create tests for the **announce_** methods you can too, but I'm ok with keeping them without unit tests.
